docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+- The same identities' data is replicated across wallets to be kept in sync
+- Interactions between identities and applications, such as authenticating, signing and verification of signatures, are mediated by the wallet
+- Components, interfaces and protocols are detailed and explained in an open specification, allowing anyone to implement their own IDM
+
+From a higher-level perspective, the Identity Manager - or IDM for short - is composed of four components: IDM Wallet, IDM Wallet UI, IDM Client and IDM Bridge:
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/diagram_components-overview.png" alt="">
+
+### IDM Wallet
+
+The IDM Wallet is similar to the physical wallet you carry everyday. It contains all the digital identities of its holder and all the information attached to them in a secure and encrypted way.
+
+Note that the IDM Wallet is an headless component, meaning it has no graphical user interface. Anyone wanting to build such GUIs, may choose from a variety of SDKs written in different programming languages, all based on the [IDM Wallet spec](idm-spec.md#idm-wallet). More information about the IDM Wallet UI can be found later on this document.
+
+There will be a reference IDM Wallet written in JavaScript, suitable to use inside a browser.
+
+#### Locker
+
+The IDM Wallet will have a locking mechanism to protect its access. A variety of locking types may be used to unlock the wallet, such as a passphrase, pattern, typeid, fingerprint, faceid, and other biometric validators. The locker holds a single secret that will be encrypted for each locking type. That single secret will be used to encrypt and decrypt all the data that lives inside the wallet.
+
+#### Managing identities
+
+Users will be able to create identities using their preferred DID-method or import existing ones. They will be guided throughout the process in the UI according to the chosen DID-method.
+
+The outcome of the creation of an identity will be a new DID, a Master Key Pair and a Device Key Pair where their correspondent public keys are listed in the DID-Document. The Master Private Key is in complete control of the DID-Document and should be used as little as possible. For that reason, it should be stored outside the IDM Wallet in a secure and recoverable way. One of those ways is via a Paper Key, where the Master Private Key is printed using machine readable representations, such as a QR code. [Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing) may also be used to split the Master Private Key into different secrets that can be shared with trustees, like family and close friends. Even if the Paper Key is lost, users may recover their Master Private Key by collecting those secrets from them.
+For the import case, a Device Key Pair will be created and its public key added to the DID-Document as well. Depending on the DID-method, the Master Private Key might be needed to update the DID-Document.

```

Comment:
Maybe replace _for the import case,_ with _when importing an identity,_

Reply:
Done!